lib/auth/dbobjectimportrule/dbobjectimportrulev1/service.go

```diff
@@ -0,0 +1,222 @@
+func (rs *DatabaseObjectImportRuleService) ListDatabaseObjectImportRules(
+	ctx context.Context, req *pb.ListDatabaseObjectImportRulesRequest,
+) (*pb.ListDatabaseObjectImportRulesResponse, error) {
+	err := rs.authorize(ctx, false, types.VerbRead, types.VerbList)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	out, next, err := rs.backend.ListDatabaseObjectImportRules(ctx, int(req.PageSize), req.PageToken)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &pb.ListDatabaseObjectImportRulesResponse{
+		Rules:         out,
+		NextPageToken: next,
+	}, nil
+}
+
+// CreateDatabaseObjectImportRule creates a new DatabaseObjectImportRule. It will return an error if the DatabaseObjectImportRule already
+// exists.
+func (rs *DatabaseObjectImportRuleService) CreateDatabaseObjectImportRule(
+	ctx context.Context, req *pb.CreateDatabaseObjectImportRuleRequest,
+) (*pb.DatabaseObjectImportRule, error) {
+	err := rs.authorize(ctx, true, types.VerbCreate)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = databaseobjectimportrule.ValidateDatabaseObjectImportRule(req.Rule)
+	if err != nil {
+		return nil, trace.Wrap(err, "validating rule")
+	}
+
+	out, err := rs.backend.CreateDatabaseObjectImportRule(ctx, req.Rule)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return out, nil
+
+}
+
+// UpsertDatabaseObjectImportRule creates a new DatabaseObjectImportRule or forcefully updates an existing DatabaseObjectImportRule.
+// This is a function rather than a method so that it can be used by the gRPC service
+// and the auth server init code when dealing with resources to be applied at startup.
+func UpsertDatabaseObjectImportRule(
+	ctx context.Context,
+	backend Backend,
+	rule *pb.DatabaseObjectImportRule,
+) (*pb.DatabaseObjectImportRule, error) {
+	if err := databaseobjectimportrule.ValidateDatabaseObjectImportRule(rule); err != nil {
+		return nil, trace.Wrap(err, "validating rule")
+	}
+	out, err := backend.UpsertDatabaseObjectImportRule(ctx, rule)
+	return out, trace.Wrap(err)
+}
+
+// UpdateDatabaseObjectImportRule updates an existing DatabaseObjectImportRule. It will throw an error if the DatabaseObjectImportRule does
+// not exist.
+func (rs *DatabaseObjectImportRuleService) UpdateDatabaseObjectImportRule(
+	ctx context.Context, req *pb.UpdateDatabaseObjectImportRuleRequest,
+) (*pb.DatabaseObjectImportRule, error) {
+	err := rs.authorize(ctx, true, types.VerbUpdate)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = databaseobjectimportrule.ValidateDatabaseObjectImportRule(req.Rule)
+	if err != nil {
+		return nil, trace.Wrap(err, "validating rule")
+	}
+
+	rule, err := rs.backend.UpdateDatabaseObjectImportRule(ctx, req.Rule)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return rule, nil
+}
+
+func (rs *DatabaseObjectImportRuleService) UpsertDatabaseObjectImportRule(ctx context.Context, req *pb.UpsertDatabaseObjectImportRuleRequest) (*pb.DatabaseObjectImportRule, error) {
```

Comment:
nit: go docs

Reply:
One day we'll have a linter for this 😅 thanks, fixed.